javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java

```diff
@@ -1504,60 +1504,22 @@ private String getPagingNextPageExpression(ClientMethod clientMethod, String met
             .replace("{context}", contextParam)
-            .replace("{serviceVersion}", getServiceVersionValue(clientMethod))
+            .replace("    .setServiceVersion({serviceVersion}))", getServiceVersionPollingStrategy(clientMethod))
             .replace("{serializerAdapter}", clientMethod.getClientReference() + ".getSerializerAdapter()")
             .replace("{intermediate-type}", clientMethod.getMethodPollingDetails().getIntermediateType().toString())
             .replace("{final-type}", clientMethod.getMethodPollingDetails().getFinalType().toString());
     }
 
     private String getSyncPollingStrategy(ClientMethod clientMethod, String contextParam) {
-        String endpoint = "null";
-        if (clientMethod.getProxyMethod() != null && clientMethod.getProxyMethod().getParameters() != null) {
-            if (clientMethod.getProxyMethod().getParameters().stream()
-                    .anyMatch(p -> p.isFromClient() && p.getRequestParameterLocation() == RequestParameterLocation.URI && "endpoint".equals(p.getName()))) {
-                // has EndpointTrait
-
-                final String baseUrl = clientMethod.getProxyMethod().getBaseUrl();
-                final String endpointReplacementExpr = clientMethod.getProxyMethod().getParameters().stream()
-                        .filter(p -> p.isFromClient() && p.getRequestParameterLocation() == RequestParameterLocation.URI)
-                        .filter(p -> baseUrl.contains(String.format("{%s}", p.getRequestParameterName())))
-                        .map(p -> String.format(".replace(%1$s, %2$s)",
-                                ClassType.String.defaultValueExpression(String.format("{%s}", p.getRequestParameterName())),
-                                p.getParameterReference()
-                        )).collect(Collectors.joining());
-                if (!CoreUtils.isNullOrEmpty(endpointReplacementExpr)) {
-                    endpoint = ClassType.String.defaultValueExpression(baseUrl) + endpointReplacementExpr;
-                }
-            }
-        }
         return clientMethod.getMethodPollingDetails().getSyncPollingStrategy()
                 .replace("{httpPipeline}", clientMethod.getClientReference() + ".getHttpPipeline()")
-                .replace("{endpoint}", endpoint)
+                .replace("    .setEndpoint({endpoint}))", getEndpointPollingStrategy(clientMethod))
                 .replace("{context}", contextParam)
-                .replace("{serviceVersion}", getServiceVersionValue(clientMethod))
+                .replace("    .setServiceVersion({serviceVersion}))", getServiceVersionPollingStrategy(clientMethod))
                 .replace("{serializerAdapter}", clientMethod.getClientReference() + ".getSerializerAdapter()")
                 .replace("{intermediate-type}", clientMethod.getMethodPollingDetails().getIntermediateType().toString())
                 .replace("{final-type}", clientMethod.getMethodPollingDetails().getFinalType().toString());
```

Comment:
Why this complexity? I kind of assume we only need to add one line?

```java
        return clientMethod.getMethodPollingDetails().getSyncPollingStrategy()
                .replace("{httpPipeline}", clientMethod.getClientReference() + ".getHttpPipeline()")
                .replace("{endpoint}", endpoint)
                .replace("{context}", contextParam)
                .replace("{serviceVersion}", getServiceVersionValue(clientMethod))
                .replace("{serializerAdapter}", clientMethod.getClientReference() + ".getSerializerAdapter()")
                .replace("{intermediate-type}", clientMethod.getMethodPollingDetails().getIntermediateType().toString())
                .replace("{final-type}", clientMethod.getMethodPollingDetails().getFinalType().toString());
                .replace(".setServiceVersion(null)", "");
```
last line

Reply:
Ah, this is better. I will update and test. Maybe `.replace("    .setServiceVersion(null)", "")` to remove the front spaces/tab as well, also`endpoint` is in the middle, and may need to handle the space and `\n`, I'm currently testing.

Reply:
Don't put the space there. Formatter should be able to handle that.

Reply:
ok, I will test.